src/gateway/http.ts

```diff
@@ -96,21 +97,13 @@ export class HTTP extends Gateway {
         httpOptions.onRequestSocketAssigned(requestParams)
       }
 
-      socket.on('lookup', () => {
-        if (httpOptions.onSocketLookup) {
-          httpOptions.onSocketLookup(requestParams)
-        }
-      })
-      socket.on('connect', () => {
-        if (httpOptions.onSocketConnect) {
-          httpOptions.onSocketConnect(requestParams)
-        }
-      })
-      socket.on('secureConnect', () => {
-        if (httpOptions.onSocketSecureConnect) {
-          httpOptions.onSocketSecureConnect(requestParams)
-        }
-      })
+      if (httpRequest.reusedSocket) {
+        return
+      }
```

Comment:
I am not sure I like how this is exiting prematurely and looking outside of its scope. We are attaching a listener on `performRequest` just to bail out based on something we already know on the outside, isn't it?

I would suggest to move this check to before attaching the on socket, does that make sense? (The tests are passing with this suggestion)

```ts
    if (!httpRequest.reusedSocket) {
      httpRequest.on('socket', (socket) => {
        if (httpOptions.onRequestSocketAssigned) {
          httpOptions.onRequestSocketAssigned(requestParams)
        }
        // ...
```

Reply:
My assumption was that `reusedSocket` property was set when the socket was assigned to the request, but I see that assumption was wrong (TIL).
My main reasoning of adding the early return inside of the **onSocket listener** was to be able to trigger the `onRequestSocketAssigned` callback. We would be skipping it on reused sockets, even though the request is new